analyzers/src/SonarAnalyzer.CSharp/SonarAnalyzer.CSharp.csproj

```
@@ -56,11 +56,19 @@
     <NoWarn>NU1605</NoWarn>
     <RunAnalyzersDuringBuild>false</RunAnalyzersDuringBuild>
   </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)' == 'Debug' ">
+    <DefineConstants>TRACE;CS</DefineConstants>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
+    <DefineConstants>TRACE;CS</DefineConstants>
+  </PropertyGroup>
```

This could be more straight forward approach. But it needs testing if it plays well with the default behavior.
```suggestion
  <PropertyGroup>
    <DefineConstants>$(DefineConstants);CS</DefineConstants>
  </PropertyGroup>
```

---

I prefer to let this as it is since it's generated by the project properties editor.